metaphor/quick_sight/extractor.py

```diff
@@ -0,0 +1,246 @@
+from metaphor.common.event_util import ENTITY_TYPES
+from metaphor.common.logger import get_logger, json_dump_to_debug_file
+from metaphor.common.utils import unique_list
+from metaphor.models.crawler_run_metadata import Platform
+from metaphor.models.metadata_change_event import AssetStructure, Chart, ChartType
+from metaphor.models.metadata_change_event import Dashboard as MetaphorDashboard
+from metaphor.models.metadata_change_event import DashboardInfo as MetaphorDashboardInfo
+from metaphor.models.metadata_change_event import (
+    DashboardLogicalID as MetaphorDashboardLogicalId,
+)
+from metaphor.models.metadata_change_event import (
+    DashboardPlatform as MetaphorDashboardPlatform,
+)
+from metaphor.models.metadata_change_event import (
+    Dataset,
+    EntityUpstream,
+    SourceInfo,
+    VirtualView,
+    VirtualViewLogicalID,
+    VirtualViewType,
+)
+from metaphor.quick_sight.cll import process_dataset_column_lineage
+from metaphor.quick_sight.config import AwsCredentials, QuickSightRunConfig
+from metaphor.quick_sight.models import Dashboard, DataSet, DataSource, ResourceType
+
+logger = get_logger()
+
+
+def create_quick_sight_client(aws: AwsCredentials) -> boto3.client:
+    return aws.get_session().client("quicksight")
+
+
+class Endpoint(enum.Enum):
+    list_dashboards = "list_dashboards"
+    list_data_sets = "list_data_sets"
+    list_data_sources = "list_data_sources"
+
+
+@dataclass
+class EndpointDictKeys:
+    list_key: str
+    item_key: str
+
+
+ENDPOINT_SETTING = {
+    Endpoint.list_data_sets: EndpointDictKeys("DataSetSummaries", "DataSetId"),
+    Endpoint.list_dashboards: EndpointDictKeys("DashboardSummaryList", "DashboardId"),
+    Endpoint.list_data_sources: EndpointDictKeys("DataSources", "DataSourceId"),
+}
+
+
+class QuickSightExtractor(BaseExtractor):
+    """QuickSight metadata extractor"""
+
+    _description = "Quick Sight metadata crawler"
+    _platform = Platform.QUICK_SIGHT
+
+    @staticmethod
+    def from_config_file(config_file: str) -> "QuickSightExtractor":
+        return QuickSightExtractor(QuickSightRunConfig.from_yaml_file(config_file))
+
+    def __init__(self, config: QuickSightRunConfig) -> None:
+        super().__init__(config)
+        self._datasets: Dict[str, Dataset] = {}
+        self._aws_config = config.aws
+        self._aws_account_id = config.aws_account_id
+
+        # Arn -> Resource
+        self._resources: Dict[str, ResourceType] = {}
+
+        # Arn -> VirtualView
+        self._virtual_views: Dict[str, VirtualView] = {}
+
+    async def extract(self) -> Collection[ENTITY_TYPES]:
+        logger.info("Fetching metadata from QuickSight")
+
+        self._client = create_quick_sight_client(self._aws_config)
+
+        self._get_resources()
+
+        self._extract_virtual_view()
+
+        entities: List[ENTITY_TYPES] = []
+        entities.extend(self._virtual_views.values())
+        entities.extend(self._extract_dashboard())
+
+        return entities
```

Comment:
```suggestion
        return self._extract_virtual_views() + self._extract_dashboards()
```

`_extract_dashboard` returns a list of dashboards, but `_extract_virtual_view` returns a None. I think we want to have both return a list of entities, and then return these two lists combined.

Reply:
sure, that sounds good.